### ISSUE TITLE
The problem of code redundancy

### ISSUE THREAD

**Issue by wanglu-cs** [Tree-Shu-Zhao/ExpandSearchProject#1]
In the run_llm_loop function, the _generate_with_gpu_padding function calls ==actor_rollout_wg.generate_sequences==. At each step, it calculates the logits values. However, in the end, it discards this and instead uses self.actor_rollout_wg.compute_log_prob(final_gen_batch_output) to recompute. I think this is quite unreasonable.
